### PR TITLE
Create Relay.md

### PR DESCRIPTION
**Pull request by ashleyhull-versent**

### CODE REVIEW

File: Running-Mastodon/Relay.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+
+
+# Nginx
+Reverse proxy the Relay via Nginx with a Let's Encrypt TLS Cert.
+
+```
+server {
+
```
Comment (nightpool) [inline Running-Mastodon/Relay.md]: no blank line

File: Running-Mastodon/Relay.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        listen 80;
+        server_name neighbours.aus.social;
+        root /home/mastodon/mastodon/public;
+        location /.well-known/acme-challenge/ {
```
Comment (nightpool) [inline Running-Mastodon/Relay.md]: add a newline before blocks

File: Running-Mastodon/Relay.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        location / {
+
+                return 301 https://$host$request_uri;
+        }
+}
+
+server {
+        listen 443 ssl http2;
+        #listen [::]:443 ssl http2;
```
Comment (nightpool) [inline Running-Mastodon/Relay.md]: uncomment

File: Running-Mastodon/Relay.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+        server_name neighbours.aus.social;
+        root /home/mastodon/mastodon/public;
+
+        # SSL
```
Comment (nightpool) [inline Running-Mastodon/Relay.md]: unnecessary comment, the fact that this is SSL configuration is obvious

File: Running-Mastodon/Relay.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        ssl_session_cache shared:SSL:10m;
+
+        ssl_certificate /etc/letsencrypt/live/neighbours.aus.social/fullchain.pem;
+        ssl_certificate_key /etc/letsencrypt/live/neighbours.aus.social/privkey.pem;
+
+        location /.well-known/acme-challenge/ {
+                allow all;
+        }
+        location / {
+
```
Comment (nightpool) [inline Running-Mastodon/Relay.md]: no blank line

File: Running-Mastodon/Relay.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        ssl_prefer_server_ciphers on;
+        ssl_session_cache shared:SSL:10m;
+
+        ssl_certificate /etc/letsencrypt/live/neighbours.aus.social/fullchain.pem;
+        ssl_certificate_key /etc/letsencrypt/live/neighbours.aus.social/privkey.pem;
+
+        location /.well-known/acme-challenge/ {
+                allow all;
+        }
+        location / {
```
Comment (nightpool) [inline Running-Mastodon/Relay.md]: needs blank line before

File: Running-Mastodon/Relay.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+                proxy_pass http://localhost:8085;
+                proxy_pass_request_headers on;  # Relay might not work without this.
+                proxy_set_header Host $http_host;  # Relay WILL not work without this.
+        }
+
```
Comment (nightpool) [inline Running-Mastodon/Relay.md]: no blank line

File: Running-Mastodon/Relay.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        ssl_certificate /etc/letsencrypt/live/neighbours.aus.social/fullchain.pem;
+        ssl_certificate_key /etc/letsencrypt/live/neighbours.aus.social/privkey.pem;
+
+        location /.well-known/acme-challenge/ {
+                allow all;
+        }
+        location / {
+
+                proxy_pass http://localhost:8085;
+                proxy_pass_request_headers on;  # Relay might not work without this.
```
Comment (nightpool) [inline Running-Mastodon/Relay.md]: Either it does or it doesn't. Which is it?

File: Running-Mastodon/Relay.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+                proxy_set_header Host $http_host;  # Relay WILL not work without this.
```
Comment (nightpool) [inline Running-Mastodon/Relay.md]: unnecessary comment (vacuously true of anything in this documentation)

File: Running-Mastodon/Relay.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+server {
+        listen 443 ssl http2;
+        #listen [::]:443 ssl http2;
+
+        server_name neighbours.aus.social;
+        root /home/mastodon/mastodon/public;
+
+        # SSL
+        ssl_protocols TLSv1.2 TLSv1.3;
```
Comment (nightpool) [inline Running-Mastodon/Relay.md]: we should remove 1.3 per our discussion on #709

**Comment by ashleyhull-versent** [comment]
Hey @nightpool 

this PR is going to be greatly expanded on so I might close this until it's closer to completed.. I'm waiting on RX14, who seems to be waiting for a response from @Gargron 

https://source.joinmastodon.org/mastodon/pub-relay/issues/9

**Comment by nightpool** [comment]
that's fine. I think it's maybe too premature to include this anyway, since
the relay isn't meant for wide deployment yet.

**Pull request closed without merging**